Formula/hledger.rb

```diff
@@ -22,54 +23,23 @@ class Hledger < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "87c5cedfe3a02b5bc598c33a64d89b947e769c0c48ce2df7c9709c9311f22fb9"
   end
 
-  depends_on "ghc" => :build
+  depends_on "ghc@9" => :build
   depends_on "haskell-stack" => :build
 
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  resource "hledger-lib" do
-    url "https://hackage.haskell.org/package/hledger-lib-1.26.1/hledger-lib-1.26.1.tar.gz"
-    sha256 "2c9ef04c2cda669e2b83b8d2d8512f5a298cd95e5f9e825e697204ba96c85da5"
-  end
-
-  resource "hledger-ui" do
-    url "https://hackage.haskell.org/package/hledger-ui-1.26.1/hledger-ui-1.26.1.tar.gz"
-    sha256 "8bdeea8383201328e85d49be823787fe7ea1daf811189a41d3d4ce8d98d47e3d"
-  end
-
-  resource "hledger-web" do
-    url "https://hackage.haskell.org/package/hledger-web-1.26.1/hledger-web-1.26.1.tar.gz"
-    sha256 "e2b0d21c6657a234d4f991439dfbbf511490c8eae09bf832c824c272c7d0e6c7"
-  end
-
   def install
-    (buildpath/"../hledger-lib").install resource("hledger-lib")
-    (buildpath/"../hledger-ui").install resource("hledger-ui")
-    (buildpath/"../hledger-web").install resource("hledger-web")
-    cd ".." do
-      system "stack", "update"
-      (buildpath/"../stack.yaml").write <<~EOS
-        resolver: lts-17.5
-        compiler: ghc-#{Formula["ghc"].version}
-        compiler-check: newer-minor
-        packages:
-        - hledger-#{version}
-        - hledger-lib
-        - hledger-ui
-        - hledger-web
-      EOS
-      system "stack", "install", "--system-ghc", "--no-install-ghc", "--skip-ghc-check", "--local-bin-path=#{bin}"
-
-      man1.install Dir["hledger-*/*.1"]
-      man5.install Dir["hledger-lib/*.5"]
-      info.install Dir["hledger-*/*.info"]
-    end
+    system "stack", "update"
+    system "stack", "install", "--system-ghc", "--no-install-ghc", "--skip-ghc-check", "--local-bin-path=#{bin}"
```

Comment:
Alternatively, I see that upstream has build steps for `cabal-install`. Might be able to do something like:
```suggestion
    system "cabal", "v2-update"
    system "cabal", "v2-install", "all:exes", *std_cabal_v2_args
```

Reply:
@cho-m as you like, but note that is not guaranteed to be reproducible, as demonstrated a few years ago when this formula randomly stopped working. I fixed it by switching from cabal to stack. To make cabal equivalently reliable requires adding a cabal project and/or freeze file, most likely targetting a stackage snapshot. Ie, jump through extra hoops to replicate stack/stackage.